lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -9,25 +9,60 @@
 #include "PassDetails.h"
 #include "circt/Dialect/FIRRTL/FIRRTLAnnotations.h"
 #include "circt/Dialect/FIRRTL/FIRRTLAttributes.h"
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.h"
 #include "circt/Dialect/FIRRTL/InstanceGraph.h"
 #include "circt/Dialect/FIRRTL/Passes.h"
+#include "circt/Support/LLVM.h"
 #include "mlir/IR/Threading.h"
 #include "llvm/ADT/APSInt.h"
+#include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/TinyPtrVector.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "firrtl-imconstprop"
 
 using namespace circt;
 using namespace firrtl;
 
-/// Return true if this is a wire or register.
+/// Return true if this is either wire, register or similar(subfield and
+/// subindex).
 static bool isWireOrReg(Operation *op) {
-  return isa<WireOp>(op) || isa<RegResetOp>(op) || isa<RegOp>(op);
+  return isa<WireOp, RegResetOp, RegOp, SubfieldOp, SubindexOp>(op);
```

Comment:
Why add subfield/subindex to this predicate?  Should the function be renamed?

Reply:
Thanks, I thought it was convenient to regards subfield/subindex as wire/reg at post processing. But it's confusing so I'll separate it.